lib/bitcoinerlab.js

```diff
@@ -0,0 +1,139 @@
+  },
+  publicKeyVerify: function (publicKey) {
+    return secp256k1.isPoint(publicKey) ? 0 : 1
+  },
+  publicKeyCreate: function (output, seckey) {
+    try {
+      const res = secp256k1.pointFromScalar(seckey, isCompressedPublicKey(output))
+      if (!res) return 2
+      output.set(res)
+      return 0
+    } catch (e) {
+      return 1
+    }
+  },
+  publicKeyConvert: function (output, pubkey) {
+    try {
+      output.set(secp256k1.pointCompress(pubkey, isCompressedPublicKey(output)))
+      return 0
+    } catch (e) {
+      if (e.toString().includes(errors.EXPECTED_POINT)) return 1
+      return 2
+    }
+  },
+  publicKeyNegate: function (output, pubkey) {
+    return elliptic.publicKeyNegate(output, pubkey)
+  },
+  publicKeyCombine: function (output, pubkeys) {
+    return elliptic.publicKeyCombine(output, pubkeys)
+  },
+  publicKeyTweakAdd: function (output, pubkey, tweak) {
+    try {
+      const res = secp256k1.pointAddScalar(pubkey, tweak, isCompressedPublicKey(output))
+      if (!res) return 2
+      output.set(res)
+      return 0
+    } catch (e) {
+      if (e.toString().includes(errors.EXPECTED_POINT)) return 1
+      return 2
+    }
+  },
+  publicKeyTweakMul: function (output, pubkey, tweak) {
+    try {
+      const res = secp256k1.pointMultiply(pubkey, tweak, isCompressedPublicKey(output))
+      if (!res) return 2
+      output.set(res)
+      return 0
+    } catch (e) {
+      if (e.toString().includes(errors.EXPECTED_POINT)) return 1
+      return 2
+    }
+  },
+  signatureNormalize: function (sig) {
+    return elliptic.signatureNormalize(sig)
+  },
+  signatureExport: function (obj, sig) {
+    return elliptic.signatureExport(obj, sig)
+  },
+  signatureImport: function (output, sig) {
+    return elliptic.signatureImport(output, sig)
+  },
+  ecdsaSign: function (obj, msg32, seckey, data, noncefn) {
+    if (noncefn) {
+      return elliptic.ecdsaSign(obj, msg32, seckey, data, noncefn)
+    }
+    try {
+      obj.signature.set(secp256k1.sign(msg32, seckey, data))
+      Object.defineProperty(obj, 'recid', {
+        get: function () {
+          const obj = { signature: this.signature, recid: null }
+          elliptic.ecdsaSign(obj, msg32, seckey, data, noncefn)
+          return obj.recid
+        }
+      })
+      return 0
+    } catch (e) {
+      if (e.toString().includes(errors.EXPECTED_PRIVATE)) return 1
+      return 2
+    }
+  },
+  ecdsaVerify: function (signature, message, publicKey) {
+    try {
+      return secp256k1.verify(message, publicKey, signature) ? 0 : 3
+    } catch (e) {
+      if (e.toString().includes(errors.EXPECTED_SIGNATURE)) return 1
+      return 2
+    }
+  },
+  ecdsaRecover: function (output, sig, recid, msg32) {
+    return elliptic.ecdsaRecover(output, sig, recid, msg32)
+  },
+  ecdh: function (output, pubkey, seckey, data, hashfn, xbuf, ybuf) {
+    return elliptic.ecdh(output, pubkey, seckey, data, hashfn, xbuf, ybuf)
```

Comment:
`@noble/secp256k1` supports ECDH: https://www.npmjs.com/package/@noble/secp256k1/v/1.7.1#getsharedsecretprivatekeya-publickeyb


However, it doesn't have any support for introspecting on the shared point, nor does it allow custom hash functions.

Still, 99.9% of people shouldn't attempt to use such features. It might be best to use `@noble/secp256k1` for the happy-path normal scenario, and fall back to elliptic if absolutely needed (or even throw an error to discourage poor choices downstream)

Reply:
I'll skip this one since I didn't find any usages. Thanks for suggestion!